File: 442-FindAllDuplicatesInAnArray/442-FindAllDuplicatesInAnArray.java
// Last updated: 6/12/2026, 5:51:45 PM
class Solution {
    public List<Integer> findDuplicates(int[] nums) 
    {
        int i = 0;

        while(i < nums.length)
        {
            int correct = nums[i] - 1;

            if(nums[i] != nums[correct])
            {
                int temp = nums[i];
                nums[i] = nums[correct];
                nums[correct] = temp;
            }
            else
            {
                i++;
            }
        }
          
        ArrayList<Integer> a = new ArrayList<>();
        for(int j = 0; j <nums.length; j++)
        {
            // System.out.println(nums[j] + "");
            if(nums[j]-1!=j)
            {
                a.add(nums[j]);
            }
            
        }
        return a;

        
    }
}

File: 442-FindAllDuplicatesInAnArray/442-FindAllDuplicatesInAnArray.py
# Last updated: 6/12/2026, 5:51:48 PM
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        hsh={}
        for n in nums:
            if n in hsh:
                hsh[n]+=1
            else:
                hsh[n]=1
        res=[]

        for key,val in hsh.items():
            if val==2:
                res.append(key)
        return res

        